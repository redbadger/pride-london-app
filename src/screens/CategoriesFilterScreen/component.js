// @flow
import React, { PureComponent } from "react";
import { View, StyleSheet } from "react-native";
import SafeAreaView from "react-native-safe-area-view";
import type { NavigationScreenProp, NavigationState } from "react-navigation";
import type { Event, EventCategoryName } from "../../data/event";
import { lightNavyBlueColor } from "../../constants/colors";
import text from "../../constants/text";
import Button from "../../components/ButtonPrimary";
import ContentPadding from "../../components/ContentPadding";
import Header from "./Header";
import List from "./List";
import locale from "../../data/locale";

export type Props = {
  navigation: NavigationScreenProp<NavigationState>,
  events: Event[],
  categories: Set<EventCategoryName>,
  toggleCategoryFilter: (Set<EventCategoryName>, string) => void,
  onClearAll: () => void
};

class CategoriesFilterScreen extends PureComponent<Props> {
  handleClearAll = () => {
    this.props.onClearAll();
  };

  applyFilters = () => {
    this.props.navigation.goBack();
  };

  handleFilterChange = (categoryLabel: string) => {
    this.props.toggleCategoryFilter(this.props.categories, categoryLabel);
  };

  render() {
    const { events, categories } = this.props;
    const buttonLabel =
      categories.size > 0 ? text.showEvents(events.length) : text.showAllEvents;

    return (
      <SafeAreaView style={styles.container}>
        <Header
          onBack={this.applyFilters}
          onClearAll={this.handleClearAll}
          selectedCategories={categories}
        />
        <View style={styles.list}>
          <List
            locale={locale}
            stagedCategories={categories}
            onPress={this.handleFilterChange}
          />
        </View>
        <View style={styles.footer}>
          <ContentPadding>
            <Button onPress={this.applyFilters} disabled={!events.length}>
              {buttonLabel}
            </Button>
          </ContentPadding>
        </View>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "space-between",
    backgroundColor: lightNavyBlueColor
  },
  list: {
    flex: 1
  },
  footer: {
    paddingVertical: 12,
    backgroundColor: lightNavyBlueColor
  }
});

export default CategoriesFilterScreen;
