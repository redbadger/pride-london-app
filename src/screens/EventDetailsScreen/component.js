// @flow
import React, { PureComponent } from "react";
import { View, StyleSheet, ScrollView } from "react-native";
import type { NavigationScreenProp } from "react-navigation";
import formatDate from "date-fns/format";
import isSameDay from "date-fns/is_same_day";
import Header from "./Header";
import IconItem from "./IconItem";
import CategoryLabel from "./CategoryLabel";
import EventMap from "./EventMap";
import Text from "../../components/Text";
import Button from "../../components/Button";
import ContentPadding from "../../components/ContentPadding";
import {
  eventDetailsBgColor,
  eventDetailsHeaderBgColor
} from "../../constants/colors";
import text from "../../constants/text";
import strings from "../../constants/strings";
import type { Event, Asset } from "../../data/event";

const locale = "en-GB";

type Props = {
  navigation: NavigationScreenProp<{ params: { eventId: String } }>,
  event: Event,
  getAssetById: string => Asset
};

const removeTimezoneFromDateString = isoString => isoString.slice(0, -6);

const renderEventOverview = event => {
  const startTime = removeTimezoneFromDateString(
    event.fields.startTime[locale]
  );
  const endTime = removeTimezoneFromDateString(event.fields.endTime[locale]);
  const dateFormat = "DD MMMM YYYY";
  const timeFormat = "HH:mm";
  const dateDisplay = isSameDay(startTime, endTime)
    ? formatDate(startTime, dateFormat)
    : `${formatDate(startTime, dateFormat)} - ${formatDate(
        endTime,
        dateFormat
      )}`;
  const timeDisplay = `${formatDate(startTime, timeFormat)} - ${formatDate(
    endTime,
    timeFormat
  )}`;

  return (
    <ContentPadding style={styles.content}>
      <Text type="h1">{event.fields.name[locale]}</Text>
      <View style={styles.categoryLabelContainer}>
        {event.fields.eventCategories[locale].map(categoryName => (
          <CategoryLabel key={categoryName} categoryName={categoryName} />
        ))}
      </View>
      <View style={styles.iconItemWrapper}>
        <IconItem
          icon={<Text type="small">icn</Text>}
          title={dateDisplay}
          content={<Text type="small">{timeDisplay}</Text>}
        />
      </View>
      <View style={styles.iconItemWrapper}>
        <IconItem
          icon={<Text type="small">icn</Text>}
          title={event.fields.locationName[locale]}
        />
      </View>
      <View style={styles.iconItemWrapper}>
        <IconItem
          icon={<Text type="small">icn</Text>}
          title={`${text.eventDetailsPrice}${
            event.fields.eventPriceLow[locale]
          }`}
        />
      </View>
      {event.fields.venueDetails &&
        event.fields.venueDetails[locale].includes(
          strings.venueDetailsGenderNeutralToilets
        ) && (
          <View style={styles.iconItemWrapper}>
            <IconItem
              icon={<Text type="small">icn</Text>}
              title={text.eventDetailsGenderNeutralToilets}
            />
          </View>
        )}
      {event.fields.accessibilityOptions &&
        event.fields.accessibilityOptions[locale].length > 0 && (
          <View style={styles.iconItemWrapper}>
            <IconItem
              icon={<Text type="small">icn</Text>}
              title={text.eventDetailsAccessibility}
              content={
                <Text type="small">
                  {event.fields.accessibilityOptions[locale].join(", ")}
                </Text>
              }
            />
          </View>
        )}
    </ContentPadding>
  );
};

const renderEventDescription = event => (
  <ContentPadding style={styles.content}>
    <Text markdown>{event.fields.eventDescription[locale]}</Text>
    <View style={styles.mapWrapper}>
      <EventMap
        lat={event.fields.location[locale].lat}
        lon={event.fields.location[locale].lon}
        locationName={event.fields.locationName[locale]}
      />
    </View>
  </ContentPadding>
);

const renderEventDetails = event =>
  (event.fields.accessibilityDetails ||
    event.fields.email ||
    event.fields.phone ||
    event.fields.ticketingUrl) && (
    <ContentPadding>
      <View style={styles.sectionDivider} />
      <View style={styles.content}>
        {event.fields.accessibilityDetails && (
          <View style={styles.detailsSection}>
            <Text type="h2">{text.eventDetailsAccessibilityDetails}</Text>
            <View style={styles.accessibilityDetailsItem}>
              <Text>{event.fields.accessibilityDetails[locale]}</Text>
            </View>
          </View>
        )}
        {(event.fields.email || event.fields.phone) && (
          <View style={styles.detailsSection}>
            <Text type="h2">{text.eventDetailsContact}</Text>
            {event.fields.email && (
              <View style={styles.contactItem}>
                <IconItem
                  icon={<Text type="small">icn</Text>}
                  title={event.fields.email[locale]}
                  titleType="text"
                />
              </View>
            )}
            {event.fields.phone && (
              <View style={styles.contactItem}>
                <IconItem
                  icon={<Text type="small">icn</Text>}
                  title={event.fields.phone[locale]}
                  titleType="text"
                />
              </View>
            )}
          </View>
        )}
        {event.fields.ticketingUrl && (
          <View style={styles.buyButton}>
            <Button
              text={text.eventDetailsBuyButton}
              url={event.fields.ticketingUrl[locale]}
            />
          </View>
        )}
      </View>
    </ContentPadding>
  );

class EventDetailsScreen extends PureComponent<Props> {
  static navigationOptions = {
    header: null
  };
  static defaultProps = {};

  render() {
    const { event, getAssetById } = this.props;
    const headerImage = getAssetById(
      event.fields.individualEventPicture[locale].sys.id
    );
    return (
      <ScrollView style={styles.container}>
        <Header
          onBackButtonPress={() => {
            this.props.navigation.goBack(null);
          }}
          imageUrl={`https:${headerImage.fields.file[locale].url}`}
        />
        {renderEventOverview(event)}
        <View style={styles.sectionDivider} />
        {renderEventDescription(event)}
        {renderEventDetails(event)}
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: eventDetailsBgColor
  },
  content: {
    flex: 1,
    paddingVertical: 15,
    backgroundColor: eventDetailsBgColor
  },
  categoryLabelContainer: {
    marginTop: 16,
    marginBottom: 20,
    flexDirection: "row",
    flexWrap: "wrap"
  },
  iconItemWrapper: {
    marginBottom: 20
  },
  sectionDivider: {
    height: 4,
    backgroundColor: eventDetailsHeaderBgColor
  },
  mapWrapper: {
    marginTop: 8
  },
  detailsSection: {
    marginBottom: 20
  },
  accessibilityDetailsItem: {
    marginTop: 8
  },
  contactItem: {
    marginTop: 16
  },
  buyButton: {
    marginTop: 16
  }
});

export default EventDetailsScreen;
