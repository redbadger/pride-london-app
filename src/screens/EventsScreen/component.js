// @flow
import React, { Component } from "react";
import { StyleSheet, View } from "react-native";
import type { NavigationScreenProp, NavigationState } from "react-navigation";
import type { SavedEvents, EventDays } from "../../data/event";
import type { FieldRef } from "../../data/field-ref";
import type { ImageSource } from "../../data/get-asset-source";
import EventList from "../../components/EventList";
import FilterHeader from "../../components/ConnectedFilterHeader";
import Loading from "../../components/Loading";
import { bgColor } from "../../constants/colors";
import {
  EVENT_DETAILS,
  EVENT_CATEGORIES_FILTER,
  FILTER_MODAL
} from "../../constants/routes";
import locale from "../../data/locale";

export type Props = {
  navigation: NavigationScreenProp<NavigationState>,
  events: EventDays,
  savedEvents: SavedEvents,
  addSavedEvent: string => void,
  removeSavedEvent: string => void,
  loading: boolean,
  refreshing: boolean,
  updateEvents: () => Promise<void>,
  getAssetSource: FieldRef => ImageSource,
  selectedCategories: Set<string>
};

class EventsScreen extends Component<Props> {
  shouldComponentUpdate = () => this.props.navigation.isFocused();

  handleFilterCategoriesPress = () => {
    this.props.navigation.navigate(EVENT_CATEGORIES_FILTER);
  };

  handleFilterButtonPress = () => {
    this.props.navigation.navigate(FILTER_MODAL);
  };

  render() {
    const {
      navigation,
      updateEvents,
      events,
      savedEvents,
      addSavedEvent,
      removeSavedEvent,
      refreshing,
      getAssetSource
    } = this.props;
    return (
      <View style={styles.container}>
        <FilterHeader
          onFilterCategoriesPress={this.handleFilterCategoriesPress}
          selectedCategories={this.props.selectedCategories}
          onFilterButtonPress={this.handleFilterButtonPress}
        />
        {this.props.loading ? (
          <Loading />
        ) : (
          <EventList
            locale={locale}
            events={events}
            savedEvents={savedEvents}
            addSavedEvent={addSavedEvent}
            removeSavedEvent={removeSavedEvent}
            refreshing={refreshing}
            onRefresh={() => {
              updateEvents();
            }}
            onPress={(eventId: string) => {
              navigation.navigate(EVENT_DETAILS, { eventId });
            }}
            getAssetSource={getAssetSource}
          />
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: bgColor
  }
});

export default EventsScreen;
