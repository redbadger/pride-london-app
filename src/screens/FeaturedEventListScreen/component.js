// @flow
import React, { PureComponent } from "react";
import { StyleSheet, View } from "react-native";
import type { NavigationScreenProp } from "react-navigation";
import type { SavedEvents, EventDays } from "../../data/event";
import type { FieldRef } from "../../data/field-ref";
import type { ImageSource } from "../../data/get-asset-source";
import EventList from "../../components/EventList";
import Header from "../../components/Header";
import { bgColor } from "../../constants/colors";
import { EVENT_DETAILS } from "../../constants/routes";
import text from "../../constants/text";

import locale from "../../data/locale";

export type Props = {
  navigation: NavigationScreenProp<{ params: { title: string } }>,
  events: EventDays,
  savedEvents: SavedEvents,
  addSavedEvent: string => void,
  removeSavedEvent: string => void,
  getAssetSource: FieldRef => ImageSource
};

class FeaturedEventsListScreen extends PureComponent<Props> {
  render() {
    return (
      <View style={styles.container}>
        <Header
          leftElement={
            <Header.BackButton
              onPress={() => {
                this.props.navigation.goBack(null);
              }}
            />
          }
          title={text.featuredEventListTitle}
        />
        <EventList
          locale={locale}
          events={this.props.events}
          savedEvents={this.props.savedEvents}
          addSavedEvent={this.props.addSavedEvent}
          removeSavedEvent={this.props.removeSavedEvent}
          onPress={(eventId: string) => {
            this.props.navigation.navigate(EVENT_DETAILS, { eventId });
          }}
          getAssetSource={this.props.getAssetSource}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: bgColor
  }
});

export default FeaturedEventsListScreen;
