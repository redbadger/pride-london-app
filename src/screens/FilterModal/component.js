// @flow
import React, { PureComponent } from "react";
import { View, StyleSheet } from "react-native";
import SafeAreaView from "react-native-safe-area-view";
import type { NavigationScreenProp } from "react-navigation";
import ShadowedScrollView from "../../components/ShadowedScrollView";
import Button from "../../components/ButtonPrimary";
import ContentPadding from "../../components/ContentPadding";
import FilterSectionList from "./FilterSectionList";
import { bgColor } from "../../constants/colors";
import tags from "../../data/tags";
import type { FilterCollection, Area } from "../../data/event-filters";
import Header from "./Header";

export type TagFilter = { [string]: Set<string> };

type Props = {
  navigation: NavigationScreenProp<{}>,
  applyButtonText: string,
  eventFilters: FilterCollection,
  numEventsSelected: number,
  numTagFiltersSelected: number,
  onChange: TagFilter => void,
  onApply: () => void,
  onCancel: () => void
};

const toggleTagFilter = (
  selectedValues: Set<string> | Set<Area>,
  sectionValue: string
) => {
  const values: Set<string> = new Set([...selectedValues]);
  if (!values.delete(sectionValue)) {
    values.add(sectionValue);
  }
  return values;
};

class FilterModal extends PureComponent<Props> {
  componentDidMount() {
    const { navigation, onCancel } = this.props;
    this.didBlurSubscription = navigation.addListener("willBlur", onCancel);
  }

  componentWillUnmount() {
    this.didBlurSubscription.remove();
  }

  didBlurSubscription: {
    remove: () => void
  };

  clearTagFilters = () =>
    this.props.onChange(
      Object.keys(tags).reduce((acc, key) => ({ ...acc, [key]: new Set() }), {})
    );

  handleCheckboxChange = (sectionName: string, sectionValue: string) => {
    this.props.onChange({
      [sectionName]: toggleTagFilter(
        this.props.eventFilters[sectionName],
        sectionValue
      )
    });
  };

  handleApplyButtonPress = () => {
    this.props.onApply();
    this.props.navigation.goBack();
  };

  render() {
    const {
      applyButtonText,
      eventFilters,
      navigation,
      numEventsSelected,
      numTagFiltersSelected
    } = this.props;
    return (
      <SafeAreaView
        style={styles.flex}
        forceInset={{ bottom: "always", top: "never" }}
      >
        <Header
          onClearPress={this.clearTagFilters}
          onCancelPress={() => navigation.goBack()}
          showClear={numTagFiltersSelected > 0}
        />
        <ShadowedScrollView style={styles.flex} shadowOpacity={0.6}>
          <FilterSectionList
            eventFilters={eventFilters}
            handleCheckboxChange={this.handleCheckboxChange}
          />
        </ShadowedScrollView>
        <View style={styles.footer}>
          <ContentPadding>
            <Button
              onPress={this.handleApplyButtonPress}
              disabled={numEventsSelected === 0}
            >
              {applyButtonText}
            </Button>
          </ContentPadding>
        </View>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  flex: {
    flex: 1,
    backgroundColor: bgColor
  },
  footer: {
    paddingVertical: 12,
    backgroundColor: bgColor
  }
});

export default FilterModal;
