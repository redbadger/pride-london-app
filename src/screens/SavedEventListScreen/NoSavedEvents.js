// @flow
import React, { PureComponent } from "react";
import {
  Image,
  PixelRatio,
  Platform,
  StyleSheet,
  ScrollView
} from "react-native";
import type { NavigationScreenProp, NavigationState } from "react-navigation";
import Text from "../../components/Text";
import ButtonPrimary from "../../components/ButtonPrimary";
import LayoutColumn from "../../components/LayoutColumn";
import ContentPadding from "../../components/ContentPadding";
import iconSave from "../../../assets/images/save.png";
import noSavedEvents from "../../../assets/images/howToSaveEvents.png";
import text from "../../constants/text";
import { EVENT_LIST } from "../../constants/routes";

type Props = {
  navigation: NavigationScreenProp<NavigationState>
};

class NoSavedEvents extends PureComponent<Props> {
  eventList = () => {
    this.props.navigation.navigate(EVENT_LIST);
  };

  render() {
    return (
      <ScrollView>
        <ContentPadding style={styles.container}>
          <LayoutColumn spacing={12}>
            <Image style={styles.image} source={noSavedEvents} />
            <Text type="h1" color="lightNavyBlueColor" style={styles.title}>
              {text.noSavedEventsTitle}
            </Text>
            <Text
              style={styles.infoText}
              accessibilityLabel={`${text.noSavedEventsPart1} ${
                text.noSavedEventsSaveButtonAlt
              } ${text.noSavedEventsPart2}`}
            >
              {text.noSavedEventsPart1}&nbsp;
              <Image style={styles.heart} source={iconSave} />
              &nbsp;{text.noSavedEventsPart2}
            </Text>
            <ButtonPrimary onPress={this.eventList}>
              {text.noSavedEventsButton}
            </ButtonPrimary>
          </LayoutColumn>
        </ContentPadding>
      </ScrollView>
    );
  }
}

// Fix for Android rendering Image inline with Text.
const heartWidth = 24 * (Platform.OS === "ios" ? 1 : PixelRatio.get());
const heartHeight = 21 * (Platform.OS === "ios" ? 1 : PixelRatio.get());

const styles = StyleSheet.create({
  container: {
    flex: 3,
    marginBottom: 12
  },
  heart: {
    width: heartWidth,
    height: heartHeight
  },
  image: {
    alignSelf: "center"
  },
  title: {
    textAlign: "center"
  },
  infoText: {
    alignSelf: "center",
    maxWidth: 300,
    textAlign: "center",
    paddingBottom: 12 // 24px between button and content
  }
});

export default NoSavedEvents;
