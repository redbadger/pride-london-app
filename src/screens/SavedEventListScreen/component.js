// @flow
import React, { PureComponent } from "react";
import { StyleSheet, View } from "react-native";
import type { NavigationScreenProp, NavigationState } from "react-navigation";
import type { SavedEvents, EventDays } from "../../data/event";
import type { FieldRef } from "../../data/field-ref";
import type { ImageSource } from "../../data/get-asset-source";
import EventList from "../../components/EventList";
import text from "../../constants/text";
import Loading from "../../components/Loading";
import Header from "../../components/Header";
import { bgColor } from "../../constants/colors";
import { EVENT_DETAILS } from "../../constants/routes";
import locale from "../../data/locale";
import NoSavedEvents from "./NoSavedEvents";

export type Props = {
  navigation: NavigationScreenProp<NavigationState>,
  events: EventDays,
  savedEvents: SavedEvents,
  addSavedEvent: string => void,
  removeSavedEvent: string => void,
  loading: boolean,
  refreshing: boolean,
  updateData: () => Promise<void>,
  getAssetSource: FieldRef => ImageSource
};

class SavedEventListScreen extends PureComponent<Props> {
  render() {
    const {
      navigation,
      updateData,
      events,
      savedEvents,
      addSavedEvent,
      removeSavedEvent,
      refreshing,
      loading,
      getAssetSource
    } = this.props;

    return (
      <View style={styles.container}>
        <Header
          title={text.savedEventsTitle}
          testID="page-heading-saved-events"
        />
        {loading && <Loading />}
        {!loading &&
          events.length === 0 && <NoSavedEvents navigation={navigation} />}
        {!loading &&
          events.length !== 0 && (
            <EventList
              locale={locale}
              events={events}
              savedEvents={savedEvents}
              addSavedEvent={addSavedEvent}
              removeSavedEvent={removeSavedEvent}
              refreshing={refreshing}
              onRefresh={() => {
                updateData();
              }}
              onPress={(eventId: string) => {
                navigation.navigate(EVENT_DETAILS, { eventId });
              }}
              getAssetSource={getAssetSource}
            />
          )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: bgColor
  }
});

export default SavedEventListScreen;
