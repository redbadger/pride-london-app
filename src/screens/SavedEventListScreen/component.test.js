// @flow
import React from "react";
import type { NavigationScreenProp, NavigationState } from "react-navigation";
import { shallow } from "enzyme";
import { SavedEventListScreen as Component } from "./component";
import EventList from "../../components/EventList";
import Loading from "../../components/Loading";
import NoSavedEvents from "./NoSavedEvents";

const navigation: NavigationScreenProp<NavigationState> = ({
  navigate: () => {}
}: any);

describe("SavedEventListScreen Component", () => {
  const event: Event = ({
    fields: {
      name: { "en-GB": "Pride in the Park" },
      location: { "en-GB": { lon: -0.12092150000000856, lat: 51.4875152 } },
      locationName: { "en-GB": "Vauxhall Pleasure Gardens" },
      startTime: { "en-GB": "2017-07-09T11:00+01:00" },
      endTime: { "en-GB": "2018-07-15T18:00+01:00" },
      eventPriceLow: { "en-GB": 19.95 },
      eventPriceHigh: { "en-GB": 30 },
      eventDescription: {
        "en-GB":
          "Lenard Pink’s singles walking tour in partnership with Pink Lobster Matchmaking, The Wild & Wonderful Women of Westminster will dissect, poke and prod those women who have battled, campaigned, fought and slept their way into the history books.\n\nPink for Pink tours will be putting the movers, shakers and shimmiers under a large microscopic lens, examining the likes of Emmeline Pankhurst, the lady who challenged the British Establishment and got the vote for women, to Nell Gwynn who became one of the most famous mistresses of Charles II and probably one of Britain’s biggest whores."
      },
      individualEventPicture: {
        "en-GB": {
          sys: {
            space: {
              sys: { type: "Link", linkType: "Space", id: "n2o4hgsv6wcx" }
            },
            id: "HNLFqItbkAmW8ssqQECIy",
            type: "Asset",
            createdAt: "2018-03-04T10:13:53.398Z",
            updatedAt: "2018-03-04T10:13:53.398Z",
            revision: 1
          }
        }
      },
      eventsListPicture: {
        "en-GB": {
          sys: {
            space: {
              sys: { type: "Link", linkType: "Space", id: "n2o4hgsv6wcx" }
            },
            id: "HNLFqItbkAmW8ssqQECIy",
            type: "Asset",
            createdAt: "2018-03-04T10:13:53.398Z",
            updatedAt: "2018-03-04T10:13:53.398Z",
            revision: 1
          }
        }
      },
      ticketingUrl: { "en-GB": "https://prideinlondon.org/pride-in-the-park/" },
      accessibilityOptions: {
        "en-GB": ["Accessible Toilets", "Hearing loop installed"]
      },
      eventCategories: {
        "en-GB": [
          "Community",
          "Social and Networking",
          "Sports and Activities",
          "Music"
        ]
      },
      recurrenceDates: { "en-GB": ["13/1/18", "15/1/18"] },
      accessibilityDetails: {
        "en-GB":
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean aliquet, odio vel imperdiet ultricies, libero nibh convallis ante, a imperdiet sapien risus non nisi. "
      },
      email: { "en-GB": "marsha.p.johnson@eventname.com" },
      phone: { "en-GB": "07891234567" },
      venueDetails: { "en-GB": ["Gender neutral toilets"] }
    }
  }: any);

  const events = ({
    entries: [event]
  }: any);

  it("renders correctly", () => {
    const output = shallow(
      <Component
        navigation={navigation}
        events={events}
        loading={false}
        refreshing={false}
        updateData={() => Promise.resolve()}
        getAssetSource={() => ({ uri: "", width: 1, height: 1 })}
        selectedCategories={new Set()}
        addSavedEvent={() => {}}
        removeSavedEvent={() => {}}
        savedEvents={new Set()}
        isFocused
      />
    );
    expect(output).toMatchSnapshot();
  });

  it("renders loading indicator when loading", () => {
    const output = shallow(
      <Component
        navigation={navigation}
        events={events}
        loading
        refreshing={false}
        updateData={() => Promise.resolve()}
        getAssetSource={() => ({ uri: "", width: 1, height: 1 })}
        selectedCategories={new Set()}
        addSavedEvent={() => {}}
        removeSavedEvent={() => {}}
        savedEvents={new Set()}
        isFocused
      />
    );

    const loadingText = output.find(Loading);

    expect(loadingText.length).toEqual(1);
  });

  it("renders no saved events screen when there are no events", () => {
    const output = shallow(
      <Component
        navigation={navigation}
        events={[]}
        loading={false}
        refreshing={false}
        updateData={() => Promise.resolve()}
        getAssetSource={() => ({ uri: "", width: 1, height: 1 })}
        selectedCategories={new Set()}
        addSavedEvent={() => {}}
        removeSavedEvent={() => {}}
        savedEvents={new Set()}
        isFocused
      />
    );

    expect(output).toMatchSnapshot();
    expect(output.find(NoSavedEvents).length).toEqual(1);
  });

  it("updates events on refresh", () => {
    const updateData = jest.fn();
    const output = shallow(
      <Component
        navigation={navigation}
        events={events}
        loading={false}
        refreshing={false}
        updateData={updateData}
        getAssetSource={() => ({ uri: "", width: 1, height: 1 })}
        selectedCategories={new Set()}
        addSavedEvent={() => {}}
        removeSavedEvent={() => {}}
        savedEvents={new Set()}
        isFocused
      />
    );

    output
      .find(EventList)
      .props()
      .onRefresh();

    expect(updateData).toHaveBeenCalled();
  });

  describe("#shouldComponentUpdate", () => {
    it("does not update when not focused", () => {
      const output = shallow(
        <Component
          navigation={navigation}
          events={events}
          loading={false}
          refreshing={false}
          updateData={() => Promise.resolve()}
          getAssetSource={() => ({ uri: "", width: 1, height: 1 })}
          selectedCategories={new Set()}
          addSavedEvent={() => {}}
          removeSavedEvent={() => {}}
          savedEvents={new Set()}
          isFocused
        />
      );
      const nextProps = {
        isFocused: false
      };

      const shouldUpdate = output.instance().shouldComponentUpdate(nextProps);

      expect(shouldUpdate).toBe(false);
    });
  });
});
