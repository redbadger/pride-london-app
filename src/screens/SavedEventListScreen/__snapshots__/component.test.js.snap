// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedEventListScreen Component renders correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Header
    testID="page-heading-saved-events"
    title="Saved Events"
  />
  <EventList
    addSavedEvent={[Function]}
    events={
      Object {
        "entries": Array [
          Object {
            "fields": Object {
              "accessibilityDetails": Object {
                "en-GB": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean aliquet, odio vel imperdiet ultricies, libero nibh convallis ante, a imperdiet sapien risus non nisi. ",
              },
              "accessibilityOptions": Object {
                "en-GB": Array [
                  "Accessible Toilets",
                  "Hearing loop installed",
                ],
              },
              "email": Object {
                "en-GB": "marsha.p.johnson@eventname.com",
              },
              "endTime": Object {
                "en-GB": "2018-07-15T18:00+01:00",
              },
              "eventCategories": Object {
                "en-GB": Array [
                  "Community",
                  "Social and Networking",
                  "Sports and Activities",
                  "Music",
                ],
              },
              "eventDescription": Object {
                "en-GB": "Lenard Pink’s singles walking tour in partnership with Pink Lobster Matchmaking, The Wild & Wonderful Women of Westminster will dissect, poke and prod those women who have battled, campaigned, fought and slept their way into the history books.

Pink for Pink tours will be putting the movers, shakers and shimmiers under a large microscopic lens, examining the likes of Emmeline Pankhurst, the lady who challenged the British Establishment and got the vote for women, to Nell Gwynn who became one of the most famous mistresses of Charles II and probably one of Britain’s biggest whores.",
              },
              "eventPriceHigh": Object {
                "en-GB": 30,
              },
              "eventPriceLow": Object {
                "en-GB": 19.95,
              },
              "eventsListPicture": Object {
                "en-GB": Object {
                  "sys": Object {
                    "createdAt": "2018-03-04T10:13:53.398Z",
                    "id": "HNLFqItbkAmW8ssqQECIy",
                    "revision": 1,
                    "space": Object {
                      "sys": Object {
                        "id": "n2o4hgsv6wcx",
                        "linkType": "Space",
                        "type": "Link",
                      },
                    },
                    "type": "Asset",
                    "updatedAt": "2018-03-04T10:13:53.398Z",
                  },
                },
              },
              "individualEventPicture": Object {
                "en-GB": Object {
                  "sys": Object {
                    "createdAt": "2018-03-04T10:13:53.398Z",
                    "id": "HNLFqItbkAmW8ssqQECIy",
                    "revision": 1,
                    "space": Object {
                      "sys": Object {
                        "id": "n2o4hgsv6wcx",
                        "linkType": "Space",
                        "type": "Link",
                      },
                    },
                    "type": "Asset",
                    "updatedAt": "2018-03-04T10:13:53.398Z",
                  },
                },
              },
              "location": Object {
                "en-GB": Object {
                  "lat": 51.4875152,
                  "lon": -0.12092150000000856,
                },
              },
              "locationName": Object {
                "en-GB": "Vauxhall Pleasure Gardens",
              },
              "name": Object {
                "en-GB": "Pride in the Park",
              },
              "phone": Object {
                "en-GB": "07891234567",
              },
              "recurrenceDates": Object {
                "en-GB": Array [
                  "13/1/18",
                  "15/1/18",
                ],
              },
              "startTime": Object {
                "en-GB": "2017-07-09T11:00+01:00",
              },
              "ticketingUrl": Object {
                "en-GB": "https://prideinlondon.org/pride-in-the-park/",
              },
              "venueDetails": Object {
                "en-GB": Array [
                  "Gender neutral toilets",
                ],
              },
            },
          },
        ],
      }
    }
    getAssetSource={[Function]}
    locale="en-GB"
    onPress={[Function]}
    onRefresh={[Function]}
    refreshing={false}
    removeSavedEvent={[Function]}
    savedEvents={Set {}}
  />
</Component>
`;

exports[`SavedEventListScreen Component renders no saved events screen when there are no events 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Header
    testID="page-heading-saved-events"
    title="Saved Events"
  />
  <NoSavedEvents
    navigation={
      Object {
        "navigate": [Function],
      }
    }
  />
</Component>
`;
